csrc/lower_validation.cpp

```diff
@@ -453,7 +474,25 @@ class VectorizeValidator : public OptInDispatch {
     // specific vectorized set.
     vectorized_set_info.word_size = (int)vector_word_size;
     vectorized_set_info.vectorized_leaf_id = v_id;
-    vectorized_set_info.vectorized_root_id = validator.vectorized_id_;
+    vectorized_set_info.vectorized_root_id = consumer_vectorized_id;
+
+    // Validate producer
+    auto pairwise_map = PairwiseRootDomainMap(producer_tv, tv);
+    auto producer_replayed_as_consumer =
+        TransformReplay::replayPasC(
+            producer_tv, tv, -1, pairwise_map, false, true)
+            .first;
+    ir_utils::TVDomainGuard domain_guard(
+        producer_tv, producer_replayed_as_consumer);
+    auto c2p_map =
+        BestEffortReplay::replayPasC(producer_tv, tv, -1, pairwise_map)
+            .getReplay();
+    if (auto c2p_it = c2p_map.find(v_id); c2p_it != c2p_map.end()) {
```

Comment:
Can this be `false`?

Reply:
I don't expect to see false. If we fail to find it in the map, we should throw an error and investigate, instead of silently doing nothing. Changed.